handlers/get-posts-by-user.go

```diff
@@ -3,69 +3,125 @@ package handlers
 	}
 
-	rows, err := db.Query("SELECT id, authorId, content, createdAt FROM posts WHERE hostId = ? ORDER BY id DESC", userID)
+	rows, err := db.Query(`
+		SELECT
+			p.id, p.authorId, p.content, p.createdAt,
+			pa.id, pa.username,
+			c.id, c.postId, c.authorId, c.content, c.createdAt,
+			ca.username
+		FROM posts p
+		JOIN users pa ON p.authorId = pa.id
+		LEFT JOIN post_comments c ON p.id = c.postId
+		LEFT JOIN users ca ON c.authorId = ca.id
+		WHERE p.hostId = ?
+		ORDER BY p.id DESC, c.createdAt DESC
+	`, hostID)
 
 	if err != nil {
-		http.Error(w, "DB error: "+err.Error(), http.StatusInternalServerError)
-		return
+		return nil, fmt.Errorf("error executing query: %w", err)
 	}
 
 	defer rows.Close()
 
-	var posts []PostWithAuthorAndHost
+	var posts []*m.FullPost
+	postsMap := make(map[int]*m.FullPost)
 
 	for rows.Next() {
-		var postID, authorID, createdAt int
-		var content string
+		var postID, postAuthorID, commentID, commentPostID, commentAuthorID sql.NullInt64
+		var postContent, postAuthorUsername, commentContent, commentAuthorUsername sql.NullString
+		var postCreatedAt, commentCreatedAt sql.NullInt64
+
+		if err := rows.Scan(
+			&postID, &postAuthorID, &postContent, &postCreatedAt,
+			&postAuthorID, &postAuthorUsername,
+			&commentID, &commentPostID, &commentAuthorID, &commentContent, &commentCreatedAt,
+			&commentAuthorUsername,
+		); err != nil {
+			return nil, fmt.Errorf("error scanning row: %w", err)
+		}
 
-		if err := rows.Scan(&postID, &authorID, &content, &createdAt); err != nil {
-			http.Error(w, "DB error: "+err.Error(), http.StatusInternalServerError)
-			return
+		post, ok := postsMap[int(postID.Int64)]
+		if !ok {
+			newPost := &m.FullPost{
+				ID:        int(postID.Int64),
+				Content:   postContent.String,
+				CreatedAt: int(postCreatedAt.Int64),
+				Author: m.DbUser{
+					ID:       int(postAuthorID.Int64),
+					BaseUser: m.BaseUser{Username: postAuthorUsername.String},
+				},
+				Host:     hostUser,
+				Comments: []m.FullComment{},
+			}
+			posts = append(posts, newPost)
+			postsMap[int(postID.Int64)] = newPost
+			post = newPost
 		}
 
-		var user DbUser
+		if commentID.Valid {
+			fullComment := m.FullComment{
+				ID:        int(commentID.Int64),
+				PostID:    int(commentPostID.Int64),
+				Content:   commentContent.String,
+				CreatedAt: commentCreatedAt.Int64,
+				Author: m.DbUser{
+					ID:       int(commentAuthorID.Int64),
+					BaseUser: m.BaseUser{Username: commentAuthorUsername.String},
+				},
+			}
+			post.Comments = append(post.Comments, fullComment)
+		}
+	}
 
-		userRow := db.QueryRow("SELECT id, userName FROM users WHERE id = ?", authorID)
+	if err := rows.Err(); err != nil {
+		return nil, fmt.Errorf("error during row iteration: %w", err)
+	}
 
-		if err := userRow.Scan(&user.ID, &user.Username); err != nil {
-			http.Error(w, "DB error: "+err.Error(), http.StatusInternalServerError)
-			return
-		}
+	finalPosts := make([]m.FullPost, len(posts))
```

Comment:
This loop, Vili. Without it, I seem to lose the order of the posts (by ID), which I did in the ORDERBY in the sql above. If I can avoid looping over all the posts again, that would be great.